Process data in models

gdumani opened the pull request:
The following thasks were performed this pull request:
- Create associations between data models
- Add testing data
- Add methods to display recent posts and comments and to increment counters

gdumani committed:
Merge pull request #1 from gdumani:models

Models

gdumani committed:
Add  associations to post and user

gdumani committed:
Fix posts association class name
Change foreign keys form string to symbols

gdumani committed:
Add undate counters and recent methods

gdumani committed:
Fix error in Posts

gdumani committed:
Fix linters

gdumani committed:
Define update conter methods as private and
execute updates after save

gdumani committed:
Correct linters

aamirkhan2478 commented on app/models/user.rb:
```diff
@@ -1,2 +1,9 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: :user_id
+  has_many :likes, foreign_key: :user_id
```
- [ ] Please replace `:user_id` with `:author_id` for `comments` and `likes` exactly as you did for posts check below :point_down:  my suggestion :slightly_smiling_face: 

```suggestion
  has_many :comments, foreign_key: :author_id
  has_many :likes, foreign_key: :author_id
```

aamirkhan2478 commented on app/models/post.rb:
```diff
@@ -1,3 +1,17 @@
+
+  private
+
+  def update_user_counter
+    author.increment!(:posts_counter)
+  end
```
[OPTIONAL] Please rename the method name with `update_post_counter` because this is for post count not for user count so it will look more understandable :slightly_smiling_face:

aamirkhan2478 commented on app/models/comment.rb:
```diff
@@ -1,4 +1,12 @@
 class Comment < ApplicationRecord
   belongs_to :user
```
- [ ] Please replace `:user` with `:author` exactly as you did in the `post.rb` file so kindly fix this issue as well :+1: check below :point_down:  my suggestion :slightly_smiling_face:  

```suggestion
  belongs_to :author, class_name: 'User', foreign_key: :author_id
```

aamirkhan2478 commented on app/models/like.rb:
```diff
@@ -1,4 +1,12 @@
 class Like < ApplicationRecord
   belongs_to :user
```
- [ ] Please replace `:user` with `:author` exactly as you did in the `post.rb` file so kindly fix this issue as well :+1: check below :point_down:  my suggestion :slightly_smiling_face:  

```suggestion
   belongs_to :author, class_name: 'User', foreign_key: :author_id
```

aamirkhan2478 commented on app/models/comment.rb:
```diff
@@ -1,4 +1,12 @@
+
+  private
+
+  def update_posts_counter
+    posts.increment!(:comments_counter)
+  end
```
[OPTIONAL] Please rename the method name with `update_comment_counter` because this is for comment count not for post count so it will look more understandable and please remove `s` from `posts.increment` :+1: check below :point_down: my suggestion :slightly_smiling_face:

 ```suggestion
  def update_comment_counter
    post.increment!(:comments_counter)
  end
```

aamirkhan2478 commented on app/models/like.rb:
```diff
@@ -1,4 +1,12 @@
+
+  private
+
+  def update_posts_counter
+    post.increment!(:likes_counter)
+  end
```
[OPTIONAL] Please rename the method name with `update_like_counter` because this is for like count not for post count so it will look more understandable :slightly_smiling_face:

aamirkhan2478 reviewed (changes requested):
Hi @gdumani, 

![](https://media2.giphy.com/media/t8YPf3ACKeouLzEIvr/giphy.gif?cid=ecf05e47k5uapdptg00nidbw3qaxx8iqgsusgaswci2c3uh1&rid=giphy.gif&ct=g)

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
✔️ Good Effort
✔️ Your design is good.
✔️ All linter passed

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

gdumani commented:
Aamir, You reviewed my code a couple of hours ago and you are requesting
some changes that are not part of the original requirements
https://github.com/gdumani/blogapp/pull/11
I checked the specifications of the code and in the original
specifications, the code indicated to user user_id instead of author_id ad
main keys. I had to create my code late because of personal issues and I
have already finished the next step of this weed, but that change means
changing all my code and migrations in PostgreSQL.
This is the diagram I was using for the project which was approved also by
the previous reviewer
https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd_v1_1.png

On Sat, Aug 19, 2023 at 4:20 PM Aamir khan ***@***.***> wrote:

> ***@***.**** requested changes on this pull request.
>
> Hi @gdumani <https://github.com/gdumani>,
>
>
> <https://camo.githubusercontent.com/9394efa3662ade294180d682ebade56334efde7f9b0ffaf2da4786d70e86ee0c/68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f74385950663341434b656f754c7a454976722f67697068792e6769663f6369643d65636630356534376b357561706470746730306e69646277337161787838697167737573676173776369326333756831267269643d67697068792e6769662663743d67>
>
> Good job so far!
> There are some issues that you still need to work on to go to the next
> project but you are almost there!
> Highlights
>
> ✔️ Good Effort
> ✔️ Your design is good.
> ✔️ All linter passed
> Required Changes ♻️
>
> *Check the comments under the review.*
> Optional suggestions
>
> *Every comment with the [OPTIONAL] prefix is not crucial enough to stop
> the approval of this PR. However, I strongly recommend you take them into
> account as they can make your code better.*
>
> Cheers and Happy coding!👏👏👏
>
> Feel free to leave any questions or comments in the PR thread if something
> is not 100% clear.
> Please, remember to tag me in your question so I can receive the
> notification.
>
> *Please, do not open a new Pull Request for re-reviews. You should use the
> same Pull Request submitted for the first review, either valid or invalid
> unless it is requested otherwise.*
> ------------------------------
>
> *As described in the Code reviews limits policy
> <https://microverse.zendesk.com/hc/en-us/articles/1500004088561> you have a
> limited number of reviews per project (check the exact number in your
> Dashboard). If you think that the code review was not fair, you can request
> a second opinion using this form <https://airtable.com/shrQAqnBwek5a0O0s>.*
> ------------------------------
>
> In app/models/user.rb
> <https://github.com/gdumani/blogapp/pull/11#discussion_r1299272785>:
>
> > +  has_many :comments, foreign_key: :user_id
> +  has_many :likes, foreign_key: :user_id
>
>
>    - Please replace :user_id with :author_id for comments and likes
>    exactly as you did for posts check below 👇 my suggestion 🙂
>
> ⬇️ Suggested change
>
> -  has_many :comments, foreign_key: :user_id
> -  has_many :likes, foreign_key: :user_id
> +  has_many :comments, foreign_key: :author_id
> +  has_many :likes, foreign_key: :author_id
>
> ------------------------------
>
> In app/models/post.rb
> <https://github.com/gdumani/blogapp/pull/11#discussion_r1299273018>:
>
> > +  def update_user_counter
> +    author.increment!(:posts_counter)
> +  end
>
> [OPTIONAL] Please rename the method name with update_post_counter because
> this is for post count not for user count so it will look more
> understandable 🙂
> ------------------------------
>
> In app/models/comment.rb
> <https://github.com/gdumani/blogapp/pull/11#discussion_r1299273385>:
>
> > @@ -1,4 +1,12 @@
>  class Comment < ApplicationRecord
>    belongs_to :user
>
>
>    - Please replace :user with :author exactly as you did in the post.rb
>    file so kindly fix this issue as well 👍 check below 👇 my suggestion 🙂
>
> ⬇️ Suggested change
>
> -  belongs_to :user
> +  belongs_to :author, class_name: 'User', foreign_key: :author_id
>
> ------------------------------
>
> In app/models/like.rb
> <https://github.com/gdumani/blogapp/pull/11#discussion_r1299273397>:
>
> > @@ -1,4 +1,12 @@
>  class Like < ApplicationRecord
>    belongs_to :user
>
>
>    - Please replace :user with :author exactly as you did in the post.rb
>    file so kindly fix this issue as well 👍 check below 👇 my suggestion 🙂
>
> ⬇️ Suggested change
>
> -  belongs_to :user
> +   belongs_to :author, class_name: 'User', foreign_key: :author_id
>
> ------------------------------
>
> In app/models/comment.rb
> <https://github.com/gdumani/blogapp/pull/11#discussion_r1299273933>:
>
> > +  def update_posts_counter
> +    posts.increment!(:comments_counter)
> +  end
>
> [OPTIONAL] Please rename the method name with update_comment_counter
> because this is for comment count not for post count so it will look more
> understandable and please remove s from posts.increment 👍 check below 👇
> my suggestion 🙂
> ⬇️ Suggested change
>
> -  def update_posts_counter
> -    posts.increment!(:comments_counter)
> -  end
> + def update_comment_counter
> +   post.increment!(:comments_counter)
> + end
>
> ------------------------------
>
> In app/models/like.rb
> <https://github.com/gdumani/blogapp/pull/11#discussion_r1299274032>:
>
> > +  def update_posts_counter
> +    post.increment!(:likes_counter)
> +  end
>
> [OPTIONAL] Please rename the method name with update_like_counter because
> this is for like count not for post count so it will look more
> understandable 🙂
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/gdumani/blogapp/pull/11#pullrequestreview-1585761977>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AONM33WRXTUAR5KOYHQMKHDXWE34XANCNFSM6AAAAAA3W2GIPA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

gdumani committed:
Apply reviewers suggetions
Change user_id to author_id in comments and likes
Create migrations for comments and likes to change previous name
rename counters in posts comments and likes

zieeco reviewed (approved):
# status: APPROVED :green_circle: :100: 

Hi, @gdumani, 


<img height="200" src="https://media0.giphy.com/media/IBd9uNMl6CNM1wDlLw/giphy.gif"/>

Good job so far 👍 Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights ⚡ 

✅ Descriptive PR 👍 
✅ Great job on implementing the suggested changes 🔥 
✅ Linters are passing 👏 


## Optional suggestions

- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
